pkgs/development/compilers/zig/0.10.0.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
```

Comment:
If we are keeping the old release, then it is not an upgrade, but an addition.

So, split this commit into two:

1. the first one will rename the old `default.nix` file to a new `0.9.1.nix` (and updating `all-packages.nix` accordingly, to keep the "buildability");
2. the second one adding Zig 0.10, with a title:

```
zig_0_10: init at 0.10.0
```